shell/client/grainview.js

```diff
@@ -319,6 +319,9 @@ GrainView.prototype.revealIdentity = function (identityId) {
       identity = _.findWhere(myIdentities, {_id: token.owner.user.identityId});
     }
   }
+  if (this._token) {
+    Meteor.call("revealIdentity", this._token, identity._id);
```

Comment:
As I understand this, the user who receives a share call this method, which causes the sharer to gain them as a contact? This seems wrong, as the sharer must trust the receiving user's client to actually call this method. I think it would make more sense instead to add some logic to `SandstormPermissions.createNewApiToken()`, populating a contact whenever the owner is `user`.